stardog/templates/post-install-add-users.yaml

```diff
@@ -0,0 +1,63 @@
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ include "stardog.fullname" . }}-add-users
+  labels:
+    app.kubernetes.io/managed-by: {{.Release.Service | quote }}
+    app.kubernetes.io/instance: {{.Release.Name | quote }}
+    app.kubernetes.io/version: {{ .Chart.AppVersion }}
+    helm.sh/chart: "{{.Chart.Name}}-{{.Chart.Version}}"
+  annotations:
+    # This is what defines this resource as a hook. Without this line, the
+    # job is considered part of the release.
+    "helm.sh/hook": post-upgrade,post-install
+    "helm.sh/hook-weight": "-5"
+    "helm.sh/hook-delete-policy": before-hook-creation
+spec:
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/managed-by: {{.Release.Service | quote }}
+        app.kubernetes.io/instance: {{.Release.Name | quote }}
+        helm.sh/chart: "{{.Chart.Name}}-{{.Chart.Version}}"
+    spec:
+      restartPolicy: Never
+      imagePullSecrets:
+      {{- if .Values.image.existingPullSecret }}
+        - name: {{ .Values.image.existingPullSecret }}
+      {{- else }}
+        - name: {{ template "stardog.fullname" . }}-pullsecret
+      {{- end }}
+      containers:
+      - name: add-users
+        image: {{ .Values.image.registry }}/{{ .Values.image.repository }}:{{ .Values.image.tag }}
+        imagePullPolicy: {{ .Values.image.pullPolicy }}
+        env:
+        - name: STARDOG_ADMIN_PW
+          valueFrom:
+            secretKeyRef:
+              name: {{ include "stardog.fullname" . }}-user-passwords
+              key: admin
+        {{- range .Values.stardog.users }}
+        - name: {{ .name }}_password
+          valueFrom:
+            secretKeyRef:
+              name: {{ include "stardog.fullname" $ }}-user-passwords
+              key: {{ .name }}
+        {{ end }}
+        command:
+        - /bin/sh
+        - -ce
+        - |
+          echo "testing stardog connectivity to http://{{ include "stardog.fullname" $ }}:5820"
+          until $(curl -o /dev/null -s -f {{ include "stardog.fullname" . }}); do sleep 1 && echo "."; done
```

Comment:
Please use port `5820` instead of `80` (on `80` it's the metrics).
I'd propose to create a env variable with the full server URL and to reuse it in all commands.

Reply:
probably you mean port 80 instead of 5820 during the availability-check?

Reply:
Right now the curl command uses port 80 (implicitly since no port is specified). Please change it to also use port 5820.
And to make the whole thing more DRY, please create an env var with the URL and use it everywhere:
```
env:
  - name: STARDOG_SERVER_URL
    value: http://{{ include "stardog.fullname" $ }}:5820
```